activesupport/test/encrypted_configuration_test.rb

```diff
@@ -42,6 +42,15 @@ class EncryptedConfigurationTest < ActiveSupport::TestCase
     assert @credentials.something[:good]
   end
 
+  test "reading example configuration" do
+    ActiveSupport::EncryptedFile.new(
```

Comment:
We already create a `ActiveSupport::EncryptedConfiguration` object in setup.  Let's reuse it. 

```ruby
@credentials.write("# comment") 
``` 

Also, `reading example configuration` seems to be confused when looking at it later (if the content of example changes, it is likely that the intention will not be transmitted).
How about `reading comment-only configuration`?

Reply:
I think we cannot use `@credentials.write("# comment")` here because `write` calls `deserialize` before actually writing to the file. That means the comment would not be written into the file but the `{}` fallback.

Reply:
`deserialize` is called only for format check of YAML, it does not affect `contents`.
https://github.com/rails/rails/blob/bfea0af4ba7d717d6a065b4370e3ccfd8869dde6/activesupport/lib/active_support/encrypted_configuration.rb#L26-L30 

This can also be confirmed by executing the following in rails console.

```ruby 
config_path = Rails.root.join("config", "credentials.yml.enc")
key_path = Rails.root.join("config", "master.key") 
credentials = ActiveSupport::EncryptedConfiguration.new(
  config_path: config_path, key_path: key_path,
  env_key: "RAILS_MASTER_KEY", raise_if_missing_key: true
) 

credentials.write("# comment") 
credentials.read  #=> "# comment"
```

Reply:
@y-yagi You are totally right. I missed that `write` accepts a YAML string. Looking at the test I thought it takes a Ruby structure (that will internally translated to YAML). Sorry about that.